File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/FinancialChart.swift
//
//  FinancialChart.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import UIKit

class FinancialChart: TKExamplesExampleViewController {

    let chart = TKChart()
    var dataPoints = [TKChartData]()

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        self.addOption("Candlestick", action: reloadChart)
        self.addOption("Ohlc", action: reloadChart)
    }

    required init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override func viewDidLoad() {
        super.viewDidLoad()
    
        chart.frame = self.view.bounds
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        chart.gridStyle.verticalLinesHidden = false
        chart.gridStyle.horizontalLinesHidden = false
        
        let data: NSArray = StockDataPoint.stockPoints()
        for i in 0..<42 {
            dataPoints.append(data[i] as! TKChartData)
        }
    
        self.reloadChart()
    }
    
    func reloadChart() {
        chart.removeAllData()
    
        var series:TKChartSeries
    
        switch self.selectedOption {
            case 0:
                series = TKChartCandlestickSeries(items: dataPoints)
            default:
                series = TKChartOhlcSeries(items: dataPoints)
                break
        }
    
        series.selection = TKChartSeriesSelection.DataPoint
        
        let yAxis = TKChartNumericAxis(minimum: 300, andMaximum: 380)
        yAxis.majorTickInterval = 20
        yAxis.style.labelStyle.textAlignment = TKChartAxisLabelAlignment(rawValue:TKChartAxisLabelAlignment.Bottom.rawValue | TKChartAxisLabelAlignment.Right.rawValue)
        yAxis.allowZoom = true
        yAxis.allowPan = true
        series.yAxis = yAxis

        chart.addSeries(series)
    
        let xAxis = chart.xAxis as! TKChartDateTimeAxis
        xAxis.minorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Days
        
        // >> chart-tick-style-set-swift
        xAxis.style.majorTickStyle.ticksOffset = -3
        xAxis.style.majorTickStyle.ticksHidden = false
        xAxis.style.majorTickStyle.ticksWidth = 1.5
        xAxis.style.majorTickStyle.ticksFill = TKSolidFill(color: UIColor(red: 203/255.0, green: 203/255.0, blue: 203/255.0, alpha: 1.0))
        xAxis.style.majorTickStyle.maxTickClippingMode = TKChartAxisClippingMode.Visible
        // << chart-tick-style-set-swift
        
        chart.yAxis!.style.labelStyle.textAlignment = TKChartAxisLabelAlignment(rawValue: TKChartAxisLabelAlignment.Bottom.rawValue | TKChartAxisLabelAlignment.Right.rawValue)
        
        // >> chart-zoom-swift
        chart.xAxis!.allowZoom = true
        chart.yAxis!.allowZoom = true
        // << chart-zoom-swift
        
        // >> chart-pan-swift
        chart.xAxis!.allowPan = true
        chart.yAxis!.allowPan = true
        // << chart-pan-swift
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}

File: Examples/Examples/Chart/ChartTypes/FinancialChart.cs
using System;
using System.Collections.Generic;

using Foundation;
using UIKit;

using TelerikUI;

namespace Examples
{
	[Register("FinancialChart")]
	public class FinancialChart: XamarinExampleViewController
	{
		TKChart chart;
		List<StockDataPoint> points;

		public override void ViewDidLoad ()
		{
			this.AddOption ("Candlestick", ReloadChart);
			this.AddOption ("Ohlc", ReloadChart);

			base.ViewDidLoad ();

			chart = new TKChart (this.View.Bounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			chart.GridStyle.VerticalLinesHidden = false;
			chart.GridStyle.HorizontalLinesHidden = false;
			this.View.AddSubview (chart);

			points = StockDataPoint.LoadStockPoints (42);

			this.ReloadChart ();
		}

		void ReloadChart()
		{
			chart.RemoveAllData ();

			TKChartSeries series = null;
			if (this.SelectedOption == 0) {
				series = new TKChartCandlestickSeries (points.ToArray ());
			} else {
				series = new TKChartOhlcSeries (points.ToArray ());
			}
			series.Selection = TKChartSeriesSelection.DataPoint;

			TKChartNumericAxis yAxis = new TKChartNumericAxis (new NSNumber (300), new NSNumber (380));
			yAxis.MajorTickInterval = 20;
			chart.YAxis = yAxis;

			chart.AddSeries (series);

			TKChartDateTimeAxis xAxis = (TKChartDateTimeAxis)chart.XAxis;
			xAxis.MinorTickIntervalUnit = TKChartDateTimeAxisIntervalUnit.Days;
			xAxis.Style.MajorTickStyle.TicksOffset = -3;
			xAxis.Style.MajorTickStyle.TicksHidden = false;
			xAxis.Style.MajorTickStyle.TicksWidth = 1.5f;
			xAxis.Style.MajorTickStyle.TicksFill = new TKSolidFill (UIColor.FromRGB (203, 203, 203));
			xAxis.Style.MajorTickStyle.MaxTickClippingMode = TKChartAxisClippingMode.Visible;


			// >> chart-zoom-cs
			xAxis.AllowZoom = true;
			yAxis.AllowZoom = true;
			// << chart-zoom-cs

			// >> chart-pan-cs
			xAxis.AllowPan = true;
			yAxis.AllowPan = true;
			// << chart-pan-cs

			yAxis.Style.LabelStyle.TextAlignment = TKChartAxisLabelAlignment.Bottom | TKChartAxisLabelAlignment.Right;



			chart.ReloadData ();
		}
	}
}

